### PR TITLE
[prodplan-no-preflight] 🛢 Allow no-preflight on plans

### PR DESCRIPTION
**Pull request by wlonk**
See https://trello.com/c/LPI2HccQ/142-as-sfdo-i-want-to-create-a-product-plan-that-has-no-preflight

cats

### CODE REVIEW

**Comment by wlonk** [comment]
@jgerigmeyer the backend piece of this should be in here, if you wanna take a look at what changes it'll take on the frontend.

All you need do is have a Plan with `preflight_flow_name = ''`.

File: metadeploy/api/serializers.py [hunk 1]
```diff
@@ -344,7 +352,9 @@ def _has_valid_steps(*, user, plan, steps, preflight):
         Every set in this method is a set of numeric Step PKs, from the
         local database.
         """
-        required_steps = set(plan.required_step_ids) - set(preflight.optional_step_ids)
+        required_steps = set(plan.required_step_ids)
+        if preflight:
+            required_steps = required_steps - set(preflight.optional_step_ids)
```
Comment (wlonk) [inline metadeploy/api/serializers.py]: `required_steps -= set(preflight.optional_step_ids)`

**Comment by jgerigmeyer** [comment]
@wlonk Looks like I missed tests too. Would you be willing to take the Python cleanup/tests?

**Comment by wlonk** [comment]
@jgerigmeyer Absolutely. It'll be a welcome change from the Redis/async tests that are so frustrating right now!

**Comment by jgerigmeyer** [comment]
@davisagli Just a reminder that this is complete, and just waiting on your sign-off. Not that you need to review every LOC, but I want to be sure it solves the problem adequately for you.

**Comment by davisagli** [comment]
@jgerigmeyer yep, I should be able to review it later today

**Pull request merged**